### PR TITLE
Delete My Data

### PR DESCRIPTION
**Pull request by mattThousand**
### Why
We'd like users to be able to delete the data stored locally on device

### This Commit
This commit adds functionality to the debug menu such that the user can delete all symptom log data stored on device, as well as the onboarding key

Oct-09-2020 16-35-41

### CODE REVIEW

File: src/Settings/DeleteConfirmation.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        expect(showMessageSpy).toHaveBeenCalledWith(
+          expect.objectContaining({
+            message: "Data deleted",
+          }),
+        )
+      })
+    })
+  })
+
+  describe("when data deletion is NOT successful", () => {
```
Comment (devinjameson) [inline src/Settings/DeleteConfirmation.spec.tsx]: Capital not necessary here?

Comment (devinjameson) [reply src/Settings/DeleteConfirmation.spec.tsx]: You could also do `when data selection fails`.

File: src/Settings/DeleteConfirmation.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            <DeleteConfirmation />
+          </SymptomLogContext.Provider>
+        </OnboardingProvider>,
+      )
+
+      fireEvent.press(getByLabelText("Delete My Data"))
+      await waitFor(() => {
+        expect(showMessageSpy).toHaveBeenCalledWith(
+          expect.objectContaining({
+            message: "Sorry, an error occured attempting to delete data",
```
Comment (devinjameson) [inline src/Settings/DeleteConfirmation.spec.tsx]: How does this look style-wise? Curious because it's long.

File: src/Settings/DeleteConfirmation.tsx [hunk 3]
```diff
@@ -0,0 +1,144 @@
+import React, { FunctionComponent } from "react"
+import { useTranslation } from "react-i18next"
+import { StatusBar, StyleSheet, View } from "react-native"
+import { showMessage } from "react-native-flash-message"
+
+import { useOnboardingContext } from "../OnboardingContext"
+import { useSymptomLogContext } from "../MyHealth/SymptomLogContext"
+
+import { Text } from "../components"
+import { TouchableOpacity } from "react-native"
```
Comment (devinjameson) [inline src/Settings/DeleteConfirmation.tsx:10]: May be worth organizing some fo these imports before merge.

File: src/Settings/DeleteConfirmation.tsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+  Spacing,
+  Buttons,
+  Typography,
+  Layout,
+  Colors,
+  Affordances,
+} from "../styles"
+
+const DeleteConfirmation: FunctionComponent = () => {
+  useStatusBarEffect("dark-content", Colors.secondary10)
```
Comment (devinjameson) [inline src/Settings/DeleteConfirmation.tsx:25]: 💯

File: src/Settings/DeleteConfirmation.tsx [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    if (deleteLogEntriesResult === SUCCESS_RESPONSE) {
+      resetOnboarding()
+      showMessage({
+        message: t("settings.data_deleted"),
+        ...Affordances.successFlashMessageOptions,
+      })
+    } else {
+      showMessage({
+        message: t("settings.errors.deleting_data"),
+        ...Affordances.successFlashMessageOptions,
```
Comment (devinjameson) [inline src/Settings/DeleteConfirmation.tsx]: Does this want the failure style option?

File: src/Settings/DeleteConfirmation.tsx [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+      })
+    }
+  }
+  return (
+    <>
+      <StatusBar backgroundColor={Colors.secondary10} />
+      <View style={style.container}>
```
Comment (devinjameson) [inline src/Settings/DeleteConfirmation.tsx:48]: We probably need a `ScrollView` in case the user makes their text really big.

File: src/Settings/DeleteConfirmation.tsx [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+  },
+  bodyText: {
+    ...Typography.body1,
+    paddingBottom: Spacing.xxxLarge,
+  },
+  subheaderText: {
+    ...Typography.header4,
+    paddingBottom: Spacing.xxSmall,
+  },
+  closeIconContainer: {
```
Comment (devinjameson) [inline src/Settings/DeleteConfirmation.tsx:113]: We're moving all this stuff into the navigation options so thankfully we won't have to do any of this moving forward. All the navigation will be handled at the Stack level.

Comment (devinjameson) [reply src/Settings/DeleteConfirmation.tsx:113]: Good for now though unless my PR gets merged before this.

File: src/Settings/index.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import { render, fireEvent } from "@testing-library/react-native"
```
Comment (devinjameson) [inline src/Settings/index.spec.tsx]:
```suggestion
import React from "react"
import { render, fireEvent } from "@testing-library/react-native"
```

File: src/Settings/index.spec.tsx [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+import React from "react"
+import Settings from "."
+import { useNavigation } from "@react-navigation/native"
+import { SettingsStackScreens } from "../navigation"
+
+jest.mock("@react-navigation/native")
+
+describe("Settings", () => {
+  describe('when the user taps "Delete All Data"', () => {
+    it("presents a confirmation modal", () => {
```
Comment (devinjameson) [inline src/Settings/index.spec.tsx]:
```suggestion
    it("presents a confirmation screen", () => {
```

File: src/navigation/SettingsStack.tsx [hunk 10]
```diff
@@ -15,20 +15,25 @@ import ExposureListDebug from "../Settings/ExposureListDebugScreen"
...
+const headerLeftDark = () => {
+  return <HeaderLeft color={Colors.white} />
 }
 
-const HeaderLeft = () => {
+const headerLeftLight = () => {
+  return <HeaderLeft color={Colors.primaryDarkBackround} />
+}
+
+const HeaderLeft = ({ color = Colors.white }) => {
```
Comment (devinjameson) [inline src/navigation/SettingsStack.tsx]: Should this be a `FunctionComponent<HeaderLeftProps>`? Also curious why `color` has a default.

File: src/navigation/SettingsStack.tsx [hunk 11]
```diff
@@ -15,20 +15,25 @@ import ExposureListDebug from "../Settings/ExposureListDebugScreen"
...
 
 import { Colors, Headers } from "../styles"
+import DeleteConfirmation from "../Settings/DeleteConfirmation"
 
 const Stack = createStackNavigator()
 
-const headerLeft = () => {
-  return <HeaderLeft />
+const headerLeftDark = () => {
+  return <HeaderLeft color={Colors.white} />
```
Comment (devinjameson) [inline src/navigation/SettingsStack.tsx]: I think I'm confused about this naming. `headerLeftDark` uses `Colors.white`?

**Review by devinjameson — APPROVED** [review]
Nice! Few comments! 👍

File: src/Settings/DeleteConfirmation.tsx (hunk 5 above)
Comment (mattThousand) [reply src/Settings/DeleteConfirmation.tsx]: it does - good catch 🕵️ 🙇

File: src/Settings/DeleteConfirmation.tsx [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+} from "../styles"
+
+const DeleteConfirmation: FunctionComponent = () => {
+  useStatusBarEffect("dark-content", Colors.secondary10)
+  const { resetOnboarding } = useOnboardingContext()
+  const { deleteAllLogEntries } = useSymptomLogContext()
+
+  const { t } = useTranslation()
+  const handleOnPressDeleteAllData = async () => {
+    const deleteLogEntriesResult = await deleteAllLogEntries()
```
Comment (johnschoeman) [inline src/Settings/DeleteConfirmation.tsx:31]: Wouldnt do this in this pr, but it might make sense to collate all of these 'delete the thing's into one spot as this view layer function might grow to be hard to follow as we add more things that need to be deleted. Having a single function to act as api would be a better separation of concerns. something like:
```typescript
  const deleteAllUserDataResult = await deleteAllUserData()
  ....
```

```typescript
  const deleteAllUserData = (): Promise<Result> => {
      return Promise.all([
           deleteThing1,
           deleteThing2,
            ....

  ....
```

File: src/locales/en.json [hunk 13]
```diff
@@ -375,7 +376,15 @@
     "share_anonymized_data_para": "Your anonymous data helps the {{healthAuthorityName}} Commission analyze, track, and contain the spread of COVID-19. Your data will be aggregated with data from other users who choose to share.\n\nYour data contains your survey responses and zipcode, but no additional or personally-identifying information.\n\n The {{healthAuthorityName}} will retain your anonymous data for a specific period of time.",
     "sharing_anonymized_data": "Sharing anonymized data",
     "stop_sharing_data": "Stop Sharing Data",
-    "you_are_sharing_data": "You are sharing anonymized data"
+    "you_are_sharing_data": "You are sharing anonymized data",
+    "delete_my_data": "Delete My Data",
+    "delete_data_disclosure1": "When you tap the \"Delete My Data\" button below, we'll remove the data that the app has stored on your phone.",
```
Comment (johnschoeman) [inline src/locales/en.json:387]:
```suggestion
    "delete_data_disclosure1": "When you tap the \"Delete My App Data\" button below, we'll remove the data that the app has stored on your phone.",
```

File: src/navigation/SettingsStack.tsx [hunk 14]
```diff
@@ -15,20 +15,29 @@ import ExposureListDebug from "../Settings/ExposureListDebugScreen"
...
 const Stack = createStackNavigator()
 
-const headerLeft = () => {
-  return <HeaderLeft />
+const headerLeftDark = () => {
+  return <HeaderLeft tintColor={Colors.white} />
 }
 
-const HeaderLeft = () => {
+const headerLeftLight = () => {
```
Comment (johnschoeman) [inline src/navigation/SettingsStack.tsx:26]: seems like a bit of an antipattern to conflate style concerns with markup. Why do we need to label these guys as `Dark` and `Light`? and you think theres a way to handle this in the style system?

**Review by johnschoeman — APPROVED** [review]
Looks good! I would wait to merge this until devs navigation refactor is in, as it'll be tough to handle the merge conflicts on that pr.
